cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
```

Comment:
> So it will work without BOOST_ALL_DYN_LINK but I think we should really figure out a way to include it, any ideas?

Using CMake list instead of space will do it I guess